numpy/core/src/multiarray/compiled_base.c

```diff
@@ -1169,7 +1179,16 @@ arr_unravel_index(PyObject *self, PyObject *args, PyObject *kwds)
     unravel_size = PyArray_MultiplyList(dimensions.ptr, dimensions.len);
 
     if (!PyArray_Check(indices0)) {
-        indices = (PyArrayObject*)PyArray_FROM_O(indices0);
+        /* ensure dtype is int-based */
+        PyArray_Descr *dtype_guess = PyArray_DescrFromType(NPY_BOOL);
+        if (dtype_guess == NULL) {
+            goto fail;
+        }
+        Py_INCREF(dtype_guess);
+        if (PyArray_DTypeFromObject(indices0, NPY_MAXDIMS, &dtype_guess) < 0) {
+            goto fail;
```

Comment:
See other comment, `PyArray_DTypeFromObject` does not fail.

Reply:
What do you mean by this comment?

Reply:
In a previous version we fed in a `dtype_guess`, so `PyArray_DTypeFromObject` would not fail. Now we feed in NULL, so the comment is out of date